contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/statsbeat_metrics/statsbeat.py

```diff
@@ -0,0 +1,146 @@
+    properties.append(LabelKey("cikey", 'customer ikey'))
+    properties.append(LabelKey("runtimeVersion", 'Python version'))
+    properties.append(LabelKey("os", 'os of application being instrumented'))
+    properties.append(LabelKey("language", 'Python'))
+    properties.append(LabelKey("version", 'version of exporter package'))
+    return properties
+
+
+class _StatsbeatMetrics:
+
+    def __init__(self, instrumentation_key):
+        self._instrumentation_key = instrumentation_key
+        self.vm_data = {}
+        self.vm_retry = True
+        # Attach metrics - metrics related to rp (resource provider)
+        self._attach_metric = LongGauge(
+            _ATTACH_METRIC_NAME,
+            'Statsbeat metric related to rp integrations',
+            'count',
+            _get_attach_properties(),
+        )
+
+    def get_metrics(self):
+        stats_metrics = []
+        if self._attach_metric:
+            attach_metric = self._get_attach_metric(self._attach_metric)
+            if attach_metric:
+                stats_metrics.append(attach_metric)
+
+        return stats_metrics
+
+    def _get_attach_metric(self, metric):
+        properties = []
+        vm_os_type = ''
+        # rpId
+        if os.environ.get("WEBSITE_SITE_NAME") is not None:
+            # Web apps
+            properties.append(LabelValue(_RP_NAMES[1]))
+            properties.append(
+                LabelValue(
+                    '{}/{}'.format(
+                        os.environ.get("WEBSITE_SITE_NAME"),
+                        os.environ.get("WEBSITE_HOME_STAMPNAME", '')),
+                )
+            )
+        elif os.environ.get("FUNCTIONS_WORKER_RUNTIME") is not None:
+            # Function apps
+            properties.append(LabelValue(_RP_NAMES[2]))
+            properties.append(LabelValue(os.environ.get("WEBSITE_HOSTNAME")))
+        elif self.vm_retry and self._get_azure_compute_metadata():
+            # VM
+            properties.append(LabelValue(_RP_NAMES[3]))
+            properties.append(
+                LabelValue(
+                    '{}//{}'.format(
+                        self.vm_data.get("vmId", ''),
+                        self.vm_data.get("subscriptionId", '')),
+                )
+            )
+            vm_os_type = self.vm_data.get("osType", '')
+        else:
+            # Not in any rp or VM metadata failed
+            properties.append(LabelValue(_RP_NAMES[4]))
+            properties.append(LabelValue(""))
+
+        properties.append(LabelValue("sdk"))  # attach type
+        properties.append(LabelValue(self._instrumentation_key))  # cikey
+        # runTimeVersion
+        properties.append(LabelValue(platform.python_version()))
+        properties.append(LabelValue(vm_os_type or platform.system()))  # os
+        properties.append(LabelValue("python"))  # language
+        # version
+        properties.append(
+            LabelValue(azure_monitor_context['ai.internal.sdkVersion']))
```

Comment:
should customMetrics handle this by default?

Reply:
@heyams I'm not sure I understand. Handle what by default? `customMetrics` is an Azure concept. `LabelValue` is an OpenCensus metrics concept.

Reply:
your telemetry client (not sure how you call it) should handle that for all telemetries to Breeze.  this is required for all telemetries sent to breeze.  I'm saying you should not do it specifically for statsbeat.

Reply:
Yes the this is already being handled [here](https://github.com/census-instrumentation/opencensus-python/blob/master/contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/__init__.py#L131) which populates the `tags` dict in breeze. The logic here in specific for stats because we also want `version` to be populated as a `custom_dimension` correct?